hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieStorageConfig.java

```diff
@@ -83,6 +84,12 @@
       .withDocumentation("Lower values increase the size of metadata tracked within HFile, but can offer potentially "
           + "faster lookup times.");
 
+  public static final ConfigProperty<String> HFILE_SCHEMA_KEY_FIELD_NAME = ConfigProperty
```

Comment:
This setting is broken because the HFileReader does not have a way to use it. Assume I specify this setting to be "someotherkey". The HFileReader will still use the hardcoded "key".

I suggest you remove this setting and all associated code and defer this for a later PR which will plug in this setting to the reader.

Reply:
This is only used for the Writer for now. Reader has it own config/constant as it doesn't have the config access. Sure, removing the config and making use of the HFileReader constant for now. In future, once we make HFileReader have config access, can introduce this config property.